File: src/Application Specific/Classes/Entity Collections/Checklist/Area/Area.cs
using System.Collections;

namespace FSM.Entity
{
    namespace Areas
    {
        public class Area
        {
            private DataTypeValidator _dataTypeValidator;

            public Area()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _AreaID = 0;

            public int AreaID
            {
                get
                {
                    return _AreaID;
                }
            }

            public object SetAreaID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AreaID", value);
                }
            }

            private string _Description = string.Empty;

            public string Description
            {
                get
                {
                    return _Description;
                }
            }

            public object SetDescription
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Description", value);
                }
            }

            private int _SectionID = 0;

            public int SectionID
            {
                get
                {
                    return _SectionID;
                }
            }

            public object SetSectionID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SectionID", value);
                }
            }

            private int _OrderNumber = 0;

            public int OrderNumber
            {
                get
                {
                    return _OrderNumber;
                }
            }

            public object SetOrderNumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderNumber", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Checklist/Area/Area.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace Areas

Public Class Area

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "Area Properties"

      
Private _AreaID As Integer = 0
Public Readonly Property AreaID() As Integer
	Get 
		Return _AreaID
	End Get	
End Property	
Public Writeonly Property SetAreaID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_AreaID", Value)
	End Set
End Property


Private _Description As String = String.Empty
Public Readonly Property Description() As String
	Get 
		Return _Description
	End Get	
End Property	
Public Writeonly Property SetDescription() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Description", Value)
	End Set
End Property


Private _SectionID As Integer = 0
Public Readonly Property SectionID() As Integer
	Get 
		Return _SectionID
	End Get	
End Property	
Public Writeonly Property SetSectionID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_SectionID", Value)
	End Set
End Property


Private _OrderNumber As Integer = 0
Public Readonly Property OrderNumber() As Integer
	Get 
		Return _OrderNumber
	End Get	
End Property	
Public Writeonly Property SetOrderNumber() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderNumber", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

